sys/aarch64/start.S

```diff
@@ -34,6 +34,11 @@ _ENTRY(_start)
 
         BL      aarch64_init
 
+        MOV     x19, x0
+        BL      aarch64_init_boot_stack
+        mov     sp, x0
+        mov     x0, x19
+
```

Comment:
Why do we need to switch to another stack?

Reply:
First stack is located in low addresses because MMU is disabled. But in `rpi3.c` MMU is enabled and KASAN is working (partially). Stack access is sanitized even if KASAN is not ready. It's hard to disable that because `rpi3.c` calls external functions and we really want to sanitize those functions. Low addresses can't be sanitized in current design so I move kernel into temporary stack in va.
We did it before for mips - #636.

Reply:
Could we return the address of the stack from `aarch64_init()`, in the same way it's done for MIPS? Currently `aarch64_init()` returns `_atags`, but it's a global buffer, so you could declare it as `extern` in `rpi3.c` and use it that way. Then, you could return the new stack pointer from `aarch64_init()`.

Reply:
done